stable/database/files/nuobackup

```diff
@@ -0,0 +1,161 @@
+#!/bin/sh
+#
+# simple script to automate backup
+#
+
+# args are:
+#   --type     - backup type: ( full | incremental | journal | get-latest )
+#   --db-name  - name of the database to backup
+#   --group    - arbitrary group to select the SMs to backup. Often the datacentre or cloud name
+#   --labels   - additional arbitrary labels to select the SMs to backup
+#   --timeout  - timeout (in seconds) to synchronously wait for the backup to complete. 0 => async backup.
+#   --backup-root - directory tree to create the backupsets inside - eg /var/opt/nuodb/backup
+
+# for debugging...
+[ -n "$NUODB_DEBUG" ] && set -x 
+
+: ${NUODB_MAX_BACKUP_HISTORY:=10}
+: ${NUODB_BACKUP_KEY:=/nuodb/nuobackup}
+
+backup_type="full"
+db_name=$DB_NAME
+backup_group=""
+labels=""
+semaphore=""
+timeout=1800
+backup_root=$BACKUP_DIR
+
+while [[ $# -gt 0 ]] && [[ "$1" == "-"* ]] ;
```

Comment:
`[[` and `]]` are not POSIX-standard facilities.  Since you say `#!/bin/sh` at the top please use POSIX test operations: this can be written `while [ $# -gt 0 ];` then in the `case` below you can check for options not starting with `-` and `break` out of the while loop

Reply:
FYI, the best way to test a script for portability is to run it with `dash`.  On a Debian/Ubuntu based system, `/bin/sh` is already `dash` and non-POSIX scripts will simply fail to run at all.  On a RedHat/CentOS based system, you'll have to install `dash` separately and invoke it directly to test.

Reply:
As always - your feedback on shell/bash scripting is much appreciated - thank you!

I have been using `set -n` as a poor-man's lint for shell scripts, so I'll try to improve that by using `dash` for that.

- Or, alternatively, I'll check whether I should just change the shebang to `/bin/bash`.